src/sink_http.c

```diff
@@ -116,44 +137,69 @@ static int check_elide(struct cb_info* info, char* full_name, double value) {
  * the callback will work to split the destination object and generate
  * a new one.
  */
-static int add_metrics(void* data,
+static int add_metrics(void *data,
                        metric_type type,
-                       char* name,
-                       void* value) {
-    struct cb_info* info = (struct cb_info*)data;
+                       char *name,
+                       void *value)
+{
+    struct cb_info *info = (struct cb_info *)data;
 
-    if (json_object_size(info->jobjects[0]) > MAX_BODY_OBJECTS) {
+    if (json_object_size(info->jobjects[0]) > MAX_BODY_OBJECTS)
+    {
         /* build an array size + 1 and copy in references */
-        json_t** jobjects = malloc(sizeof(json_t*) * (info->jobjects_count + 1));
-        memcpy(&jobjects[1], &info->jobjects[0], sizeof(json_t*) * info->jobjects_count);
+        json_t **jobjects = malloc(sizeof(json_t *) * (info->jobjects_count + 1));
+        memcpy(&jobjects[1], &info->jobjects[0], sizeof(json_t *) * info->jobjects_count);
         free(info->jobjects);
 
         info->jobjects = jobjects;
         info->jobjects[0] = json_object();
         info->jobjects_count++;
     }
-    json_t* obj = info->jobjects[0];
-    const statsite_config* config = info->config;
+    json_t *obj = info->jobjects[0];
+    const statsite_config *config = info->config;
 
+    if (info->sink->filter_regex != NULL)
+    {
+        pcre2_match_data *match_data = pcre2_match_data_create_from_pattern(info->sink->filter_regex, NULL);
+
+        int rc = pcre2_match(
+            info->sink->filter_regex,
+            (const unsigned char *)name,
+            strlen(name),
+            0, /* start at offset 0 in the subject */
+            0, /* default options */
+            match_data,
+            NULL); /* use default match context */
+
+        pcre2_match_data_free(match_data);
+
+        /* We have a match and should do nothing */
```

Comment:
If the regex is to filter which metrics to send shouldn't we early exit if there was no match and continue if there was a match?

Reply:
Yeah I had this backwards, stand by